frontend/src/views/ChatView/ChatView.tsx

```diff
@@ -1,9 +1,153 @@
 import styles from "./ChatView.module.css";
+import { useEffect, useState, useContext } from "react";
+import { socket } from "../../context/socket";
+import { useParams } from "react-router-dom";
+import { socketContext } from "../../context/SocketContext";
+import { MessageBody } from "../../types/MessageBody";
 
 const ChatView = () => {
+	const [messages, setMessages] = useState<MessageBody[]>([]);
+	const { setCurrentChat, currentUser } = useContext(socketContext);
+	const { chatId } = useParams();
+
+	useEffect(() => {
+		const getData = (message: string, username: string) => {
+			const receivedChatSound = new Audio("/livechat-129007.mp3");
+
+			setMessages((prev) => {
+				const messageBody = {
+					message: message,
+					username: username,
+					needsLabel:
+						prev.length >= 1
+							? prev[prev.length - 1].username !== username
```

Comment:
there's a cool new method to make this cleaner:


```suggestion
							? prev.at(-1).username !== username
```